eventuate-local-java-cdc-connector-mysql-binlog/src/main/java/io/eventuate/local/mysql/binlog/OffsetProcessor.java

```diff
@@ -0,0 +1,88 @@
+package io.eventuate.local.mysql.binlog;
+
+import io.eventuate.common.eventuate.local.BinlogFileOffset;
+import io.eventuate.local.db.log.common.OffsetStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class OffsetProcessor {
+  protected Logger logger = LoggerFactory.getLogger(getClass());
+
+  private ConcurrentLinkedQueue<FutureOffset> futureOffsets = new ConcurrentLinkedQueue<>();
+  private AtomicBoolean processingOffsets = new AtomicBoolean(false);
+  private OffsetStore offsetStore;
+
+  public OffsetProcessor(OffsetStore offsetStore) {
+    this.offsetStore = offsetStore;
+  }
+
+  public void saveOffset(CompletableFuture<?> future, BinlogFileOffset binlogFileOffset) {
+    futureOffsets.add(new FutureOffset(future, binlogFileOffset));
+
+    future.whenComplete((o, throwable) -> processOffsets());
```

Comment:
`processOffsets()` should only be called if future is successful

Reply:
Chris, it just starts checking future queue.
If some future in queue is completed with exception - processing stopped.
There is a method: checkFutureForException

So, if start processing only if future succeed, it does not mean that other future
will not try to process it, anyway we need this check (checkFutureForException).
But there is no reason to check it twice, I think.